ravenframework/Metrics/metrics/DTW.py

```diff
@@ -126,49 +126,53 @@ def run(self, x, y, weights=None, axis=0, **kwargs):
     value = self.dtwDistance(X, Y)
     return value
 
-  def dtwDistance(self, x, y):
+  def dtwDistance(self, x, y, returnPath=False):
     """
       This method actually calculates the distance between two histories x and y
       @ In, x, numpy.ndarray, data matrix for x
       @ In, y, numpy.ndarray, data matrix for y
       @ Out, value, float, distance between x and y
     """
     r, c = len(x[0,:]), len(y[0,:])
-    D0 = np.zeros((r + 1, c + 1))
+    D0 = np.zeros((r+1, c+1))
     D0[0, 1:] = np.inf
     D0[1:, 0] = np.inf
-    D1 = D0[1:, 1:]
     D1 = spatialDistance.cdist(x.T,y.T, metric=self.localDistance)
-    C = D1.copy()
-    for i in range(r):
-      for j in range(c):
-        D1[i, j] += min(D0[i, j], D0[i, j+1], D0[i+1, j])
-    if len(x)==1:
-      path = np.zeros(len(y)), range(len(y))
-    elif len(y) == 1:
-      path = range(len(x)), np.zeros(len(x))
+    D0[1:, 1:] = D1
+
+    # Populate the distance matrix
+    for i in range(1, r+1):
+        for j in range(1, c+1):
+            D0[i, j] += min(D0[i-1, j], D0[i, j-1], D0[i-1, j-1])
```

Comment:
It seems to me this is different from previous implementation. The D0[i, j] will be updated while in previous calculation D1 is used, which means updated value will not be used in next distance calculation. Does this means there is a bug in our previous implementation?